### ISSUE TITLE
DataFlow analysis requires both named and underlying ItemX fields be assigned

### ISSUE THREAD

**Issue by VSadov** [dotnet/roslyn#13046]
The following should not be an error. 

```
        static void Main(string[] args)
        {
            (string A, string B) ss;

            ss.A = "q";
            ss.Item2 = "w";

            System.Console.WriteLine(ss);
        }
```

Severity	Code	Description	Project	File	Line	Suppression State
Error	CS0165	Use of unassigned local variable 'ss'	Program.cs	18	Active

**Comment by VSadov** [comment]
Note that the bug exists in both C# and VB for the same reasons and the fix be implemented for both languages.

One of possible results of bad def assignment analysis could be wrong async captures (since that relies on unassignendness at suspend points). 
When fixing, Check that all is ok when tuple locals are captured.

**Comment by jcouv** [comment]
Fixed by #13107
